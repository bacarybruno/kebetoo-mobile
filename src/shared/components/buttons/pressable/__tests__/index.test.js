import { TouchableNativeFeedback } from 'react-native'

import setupTest from '@app/config/jest-setup'

import Pressable, { CustomPressable } from '../index'

const givenPressable = setupTest(Pressable)({
  borderless: true,
  foreground: false,
})

it('renders Pressable', () => {
  const { wrapper } = givenPressable()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('renders a CustomPressable component for old versions', () => {
  const { wrapper } = givenPressable({
    platform: {
      OS: 'android',
      Version: 1,
      select: (objs) => objs.default,
    },
  })
  expect(wrapper.root.findAllByType(CustomPressable).length).toBe(1)
})

it('renders a CustomPressable component for newer versions', () => {
  const { wrapper } = givenPressable({
    platform: {
      OS: 'ios',
      Version: 50,
      select: (objs) => objs.default,
    },
  })
  expect(wrapper.root.findAllByType(TouchableNativeFeedback).length).toBe(1)
})
