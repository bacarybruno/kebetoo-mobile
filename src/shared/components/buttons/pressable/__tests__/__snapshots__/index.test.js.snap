// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pressable 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
/>
`;
